### PR TITLE
Add 3.11 to testing matrix and a new linter workflow

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**

### CODE REVIEW

File: src/taipy/core/_scheduler/_scheduler.py
Comment (jrobinAV) [inline src/taipy/core/_scheduler/_scheduler.py]: mkdocs expects a new line to have one more tab indentation so it can interpret it as the following of the previous line. 
Was it automatic formatting? If so, this is a problem we need to fix.
```suggestion
             wait (bool): Wait for the scheduled jobs created from the pipeline submission to be finished in
                 asynchronous mode.
             timeout (Union[float, int]): The optional maximum number of seconds to wait for the jobs to be finished
                 before returning.
```

File: src/taipy/core/_scheduler/_scheduler.py
Comment (jrobinAV) [inline src/taipy/core/_scheduler/_scheduler.py]:
```suggestion
             wait (bool): Wait for the scheduled job created from the task submission to be finished in asynchronous
             mode.
                 timeout (Union[float, int]): The optional maximum number of seconds to wait for the job to be finished
                 before returning.
```

File: src/taipy/core/config/checkers/_data_node_config_checker.py
Comment (jrobinAV) [inline src/taipy/core/config/checkers/_data_node_config_checker.py]:
```suggestion
                f'must be either "pandas", "modin", "numpy" or a custom type.',
```

File: src/taipy/core/config/data_node_config.py
Comment (jrobinAV) [inline src/taipy/core/config/data_node_config.py:37]:
```suggestion
                "in_memory".
```

File: src/taipy/core/config/data_node_config.py
Comment (jrobinAV) [inline src/taipy/core/config/data_node_config.py]:
```suggestion
                    parameters which are also
```

File: src/taipy/core/config/data_node_config.py
Comment (jrobinAV) [inline src/taipy/core/config/data_node_config.py]:
```suggestion
                _False_.
```

File: src/taipy/core/config/data_node_config.py
Comment (jrobinAV) [inline src/taipy/core/config/data_node_config.py]:
```suggestion
                _"postgresql"_.
```

File: src/taipy/core/config/data_node_config.py
Comment (jrobinAV) [inline src/taipy/core/config/data_node_config.py]:
```suggestion
                _False_.
```

File: src/taipy/core/config/data_node_config.py
Comment (jrobinAV) [inline src/taipy/core/config/data_node_config.py]:
```suggestion
                _"postgresql"_.
```

File: src/taipy/core/config/data_node_config.py
Comment (jrobinAV) [inline src/taipy/core/config/data_node_config.py]:
```suggestion
                list of SQL queries.
```

File: src/taipy/core/config/data_node_config.py
Comment (jrobinAV) [inline src/taipy/core/config/data_node_config.py]:
```suggestion
            custom_document (Any): The custom document class to store, encode, and decode data when reading and writing
                to a Mongo collection. The custom_document can have optional `decode` method to decode data in the Mongo 
                collection to a custom object, and `encode` method to encode the object's properties to the Mongo 
                collection when writing.
```

File: src/taipy/core/config/data_node_config.py
Comment (jrobinAV) [inline src/taipy/core/config/data_node_config.py]:
```suggestion
                string.
```

File: src/taipy/core/data/sql.py
Comment (jrobinAV) [inline src/taipy/core/data/sql.py]:
```suggestion
            the current version is used.
```

File: src/taipy/core/data/sql_table.py
Comment (jrobinAV) [inline src/taipy/core/data/sql_table.py]:
```suggestion
            the current version is used.
```

File: src/taipy/core/job/job.py
Comment (jrobinAV) [inline src/taipy/core/job/job.py]:
```suggestion
            the latest version is used.
```

File: src/taipy/core/pipeline/pipeline.py
Comment (jrobinAV) [inline src/taipy/core/pipeline/pipeline.py]:
```suggestion
            the latest version is used.
```

File: src/taipy/core/pipeline/pipeline.py
Comment (jrobinAV) [inline src/taipy/core/pipeline/pipeline.py]:
```suggestion
            wait (bool): Wait for the scheduled jobs created from the pipeline submission to be finished in asynchronous
                mode.
            timeout (Union[float, int]): The maximum number of seconds to wait for the jobs to be finished before
                returning.
```

File: src/taipy/core/scenario/scenario.py
Comment (jrobinAV) [inline src/taipy/core/scenario/scenario.py]:
```suggestion
            the latest version is used.
```

File: src/taipy/core/scenario/scenario.py
Comment (jrobinAV) [inline src/taipy/core/scenario/scenario.py]:
```suggestion
            wait (bool): Wait for the scheduled jobs created from the scenario submission to be finished in
                asynchronous mode.
            timeout (Union[float, int]): The optional maximum number of seconds to wait for the jobs to be finished
                before returning.
```

File: src/taipy/core/taipy.py
Comment (jrobinAV) [inline src/taipy/core/taipy.py]:
```suggestion
            returning.
```

File: src/taipy/core/taipy.py
Comment (jrobinAV) [inline src/taipy/core/taipy.py]:
```suggestion
        If the scenario is primary, it can only be deleted if it is the only scenario in the cycle. In that case, its 
        cycle is also deleted.
```

File: src/taipy/core/task/task.py
Comment (jrobinAV) [inline src/taipy/core/task/task.py]:
```suggestion
            latest version is used.
```

File: src/taipy/core/task/task.py
Comment (jrobinAV) [inline src/taipy/core/task/task.py]:
```suggestion
            wait (bool): Wait for the scheduled job created from the task submission to be finished in asynchronous
                mode.
            timeout (Union[float, int]): The maximum number of seconds to wait for the job to be finished before
                returning.
```

**Review by jrobinAV — COMMENTED** [review]
- Several docstrings were wrongly formatted due to too long lines. I hope it s not automatic formatting. Otherwise, it is very error-prone.

- The `# noqa: E722` comment is a code smell to me. I created a [ticket](https://github.com/Avaiga/taipy-core/issues/432) to fix it in the future. It s really not a priority though.

File: src/taipy/core/config/data_node_config.py
Comment (joaoandre-avaiga) [reply src/taipy/core/config/data_node_config.py:37]: I think in_memory indentation is correct. It is on the same level as the line above which is already idented in relation to the attribute.

**Comment by joaoandre-avaiga** [comment]
Hey guys, I fix most of the issues with pycodestyle and mypy. I've also corrected and double check the docstrings for indentation erros that can cause problems on mkdocs, but I would still like for you guys to take a look to make sure I didn't missed anything.

There things we should review after the release:
 - Lines with # noqa, means that we are ignoring pycodestyle errors.
 - Lines with # type: ignore, means we are ignoring mypy errors.

Some mypy errors on the repositories will require the refactor that we talked about doing. It is important we try to keep this in mind in the future because those errors can indicate some design flaw in a feature. After this PR is merged, a new action will check some linters and will break the pipeline if the linters don't pass.

**Review by jrobinAV — APPROVED** [review]

File: src/taipy/core/config/data_node_config.py
Comment (jrobinAV) [reply src/taipy/core/config/data_node_config.py:37]: right

**Comment by jrobinAV** [comment]
Sounds good to me. I would like to merge it as soon as possible, since it could impact the merge of other branches.

**Pull request merged**
